compiler-plugin/src/main/java/io/ballerina/stdlib/http/compiler/HttpResourceValidator.java

```diff
@@ -899,6 +906,14 @@ private static boolean isAllowedQueryParamType(TypeDescKind kind, TypeSymbol typ
                 kind == TypeDescKind.DECIMAL || kind == TypeDescKind.BOOLEAN;
     }
 
+    private static boolean isEnumQueryParamType(TypeSymbol typeSymbol) {
+        if (typeSymbol.typeKind() == TypeDescKind.TYPE_REFERENCE) {
+            TypeReferenceTypeSymbol referenceTypeSymbol = (TypeReferenceTypeSymbol) typeSymbol;
+            return referenceTypeSymbol.definition().kind() == SymbolKind.ENUM;
```

Comment:
Should not we check the referred type of the enum?

Reply:
Updated the logic